src/engine/x86-64/X86_64Interpreter.v3

```diff
@@ -2197,6 +2202,121 @@ class X86_64InterpreterGen(ic: X86_64InterpreterCode, w: DataWriter) {
 			asm.q.sub_r_i(r_vsp, valuerep.slot_size);
 			endHandler();
 		}
+		for (t in [
+			(Opcode.I8X16_NE, asm.pcmpeqb_s_s),
+			(Opcode.I16X8_NE, asm.pcmpeqw_s_s),
+			(Opcode.I32X4_NE, asm.pcmpeqd_s_s),
+			(Opcode.I64X2_NE, asm.pcmpeqq_s_s)
+		]) {
+			bindHandler(t.0);
+			asm.movdqu_s_m(r_xmm0, vsph[-2].value);
+			asm.movdqu_s_m(r_xmm1, vsph[-1].value);
+			t.1(r_xmm0, r_xmm1);
+			t.1(r_xmm1, r_xmm1);
+			asm.xorps_s_s(r_xmm0, r_xmm1);
+			asm.movdqu_m_s(vsph[-2].value, r_xmm0);
+			decrementVsp();
+			endHandler();
+		}
+		var GREATER = true;
+		var LESS = false;
+		for (t in [
+			(Opcode.I8X16_GT_S, asm.pcmpgtb_s_s, GREATER),
+			(Opcode.I16X8_GT_S, asm.pcmpgtw_s_s, GREATER),
+			(Opcode.I32X4_GT_S, asm.pcmpgtd_s_s, GREATER),
+			(Opcode.I64X2_GT_S, asm.pcmpgtq_s_s, GREATER),
+
+			(Opcode.I8X16_LT_S, asm.pcmpgtb_s_s, LESS),
+			(Opcode.I16X8_LT_S, asm.pcmpgtw_s_s, LESS),
+			(Opcode.I32X4_LT_S, asm.pcmpgtd_s_s, LESS),
+			(Opcode.I64X2_LT_S, asm.pcmpgtq_s_s, LESS)
+		]) {
+			bindHandler(t.0);
+			if (t.2== GREATER) {
```

Comment:
I think it makes sense to factor out a method to load `r_xmm0` and `r_xmm1` at this point. For the comparisons where the order is reversed, another method that loads them in the other order.